### PR TITLE
#87 with local trick

### PR DESCRIPTION
**Pull request by Delfic**
Changed the unsafe for unchecked in the unit test since Idea doesn't "detect" that unsafe is unused.
I found this local inspection creation while debugging the code of why the global inspection stuff did nothing. Need help filling in those two `null` in the supplier since I don't really know if we can get them at that point of the execution.

### CODE REVIEW

File: src/main/java/com/dubreuia/processors/java/inspection/RedundantSuppressionSupplier.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.function.Supplier;
+
+public class RedundantSuppressionSupplier implements Supplier<LocalInspectionTool> {
+    @Override
+    public LocalInspectionTool get() {
+        return (new RedundantSuppressInspection()).
+                createLocalTool(
+                        (RedundantSuppressionDetector) LanguageInspectionSuppressors.INSTANCE.forLanguage(JavaLanguage.INSTANCE)
+                        , null
+                        , null);
```
Comment (Delfic) [inline src/main/java/com/dubreuia/processors/java/inspection/RedundantSuppressionSupplier.java:18]: Need help filling these in.
This implementation removes nothing, neither used or unused so I would love to try it with these properly passed in.

File: src/main/java/com/dubreuia/processors/java/JavaProcessor.java [hunk 2]
```diff
@@ -100,7 +103,13 @@
...
 
     suppressAnnotation(Action.suppressAnnotation,
-            SuppressionAnnotationInspection::new),
+            () -> (new RedundantSuppressInspection())
+                    .createLocalTool(
+                            (RedundantSuppressionDetector)
+                                    LanguageInspectionSuppressors.INSTANCE
+                                            .forLanguage(JavaLanguage.INSTANCE)
+                            , null
+                            , null)),
```
Comment (Delfic) [inline src/main/java/com/dubreuia/processors/java/JavaProcessor.java:112]: Need help filling in these. This might be the only thing left.

**Comment by dubreuia** [comment]
Hey @Delfic thank you for the PR I appreciate it. I'll look into it. And there's a test failure if you can look at it.

**Comment by Delfic** [comment]
> Hey @Delfic thank you for the PR I appreciate it. I'll look into it. And there's a test failure if you can look at it.

They're failing "by design" as I've changed them to test for #87 That's why this is a draft PR, I couldn't quite finish.
You should have a look at the discussion there.

There's a bunch of weird stuff in the implementation of the `RedundantSuppressionInspection` one was that it would not find `@SuppressWarnings` but only `@java.lang.SuppressWarnings` on the files.
That's what made me install the plug-in locally and it still failed.

I suggest you change the name of the config to "Remove all suppress warnings" rather than the unused as this might take more than a few years (as it already has)

**Pull request closed without merging**